src/VisualStudio/Core/Def/Options/VisualStudioOptionPersisterProvider.cs

```diff
@@ -73,23 +65,23 @@ private async Task<IOptionPersister> CreatePersisterAsync(CancellationToken canc
 
     private void RefreshOption(OptionKey2 optionKey, object? newValue)
     {
-        if (_legacyGlobalOptions.GlobalOptions.RefreshOption(optionKey, newValue))
+        if (_legacyGlobalOptions.Value.GlobalOptions.RefreshOption(optionKey, newValue))
         {
             // We may be updating the values of internally defined public options.
             // Update solution snapshots of all workspaces to reflect the new values.
-            _legacyGlobalOptions.UpdateRegisteredWorkspaces();
+            _legacyGlobalOptions.Value.UpdateRegisteredWorkspaces();
         }
     }
 
     /// <summary>
     /// Returns a service without doing a transition to the UI thread to cast the service to the interface type. This should only be called for services that are
     /// well-understood to be castable off the UI thread, either because they are managed or free-threaded COM.
     /// </summary>
-    private async ValueTask<I?> GetFreeThreadedServiceAsync<T, I>() where I : class
+    private I? GetFreeThreadedService<T, I>() where I : class
```

Comment:
This can be done as a follow-up, but do we need this extension anymore? If all the underlying helpers from the shell are now smarter, maybe we can delete it.